transport.go

```diff
@@ -113,25 +123,35 @@ func (t *transport) Dial(ctx context.Context, raddr ma.Multiaddr, p peer.ID) (tp
 		return nil, err
 	}
 	tlsConf, keyCh := t.identity.ConfigForPeer(p)
-	sess, err := quic.DialContext(ctx, pconn, addr, host, tlsConf, quicConfig)
+	pconn, err := t.connManager.Dial(network, udpAddr)
 	if err != nil {
 		return nil, err
 	}
-	localMultiaddr, err := toQuicMultiaddr(sess.LocalAddr())
+	sess, err := quic.DialContext(ctx, pconn, addr, host, tlsConf, quicConfig)
 	if err != nil {
+		pconn.DecreaseCount()
 		return nil, err
 	}
-
 	// Should be ready by this point, don't block.
 	var remotePubKey ic.PubKey
 	select {
 	case remotePubKey = <-keyCh:
 	default:
 	}
 	if remotePubKey == nil {
+		pconn.DecreaseCount()
 		return nil, errors.New("go-libp2p-quic-transport BUG: expected remote pub key to be set")
 	}
+	go func() {
+		<-sess.Context().Done()
+		pconn.DecreaseCount()
+	}()
```

Comment:
We can leave this for now but we may want to simply wrap close in the future. Leaving an extra goroutine open can get expensive.

Reply:
Not every connection is closed by calling `Close`. Only the peer that's actively closing the connection calls `Close`. The other peer will just receive a connection failure. Or the connection might time out, in which case nobody calls `Close` at all.